### PR TITLE
Update db_crashtest.py for support for dir creation on remote storage

### PR DESCRIPTION
**Pull request by akankshamahajan15**
Summary: 
- add TEST_TMPDIR_EXPECTED env in crash test if expected dir is on different filesystem. If TEST_TMPDIR_EXPECTED is not specified, it'll fallback to default value of TEST_TMPDIR (Same as before)

Test Plan: Ran locally

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: db_stress_tool/db_stress_gflags.cc [hunk 1]
```diff
@@ -598,6 +598,8 @@ DEFINE_bool(mock_direct_io, false,
 
 DEFINE_bool(statistics, false, "Create database statistics");
 
+DEFINE_string(statistics_string, "", "Serialized statistics string");
```
Comment (anand1976) [inline db_stress_tool/db_stress_gflags.cc:601]: Description is a little confusing. This is for creation of a 3rd party statistics object right?

File: tools/db_crashtest.py [hunk 2]
```diff
@@ -233,7 +235,10 @@ def get_dbname(test_name):
             print("Running DB cleanup command - %s\n" % cleanup_cmd)
             # Ignore failure
             os.system(cleanup_cmd)
-        os.mkdir(dbname)
+        try:
```
Comment (anand1976) [inline tools/db_crashtest.py:238]: Is this for the case where the test dir is created on remote storage and we don't care whether this step succeeds or not?

File: tools/db_crashtest.py [hunk 3]
```diff
@@ -245,7 +250,12 @@ def setup_expected_values_dir():
     if expected_values_dir is not None:
         return expected_values_dir
     expected_dir_prefix = "rocksdb_crashtest_expected_"
-    test_tmpdir = os.environ.get(_TEST_DIR_ENV_VAR)
+    test_tmpdir = os.environ.get(_TEST_EXPECTED_DIR_ENV_VAR)
```
Comment (anand1976) [inline tools/db_crashtest.py:253]: This code block is a bit confusing. Can you change test_tmpdir to test_exp_tmpdir or something similar?

File: tools/db_crashtest.py [hunk 4]
```diff
@@ -265,7 +275,12 @@ def setup_multiops_txn_key_spaces_file():
     if multiops_txn_key_spaces_file is not None:
         return multiops_txn_key_spaces_file
     key_spaces_file_prefix = "rocksdb_crashtest_multiops_txn_key_spaces"
-    test_tmpdir = os.environ.get(_TEST_DIR_ENV_VAR)
+    test_tmpdir = os.environ.get(_TEST_EXPECTED_DIR_ENV_VAR)
```
Comment (anand1976) [inline tools/db_crashtest.py:278]: Change test_tmpdir variable name?

File: tools/db_crashtest.py (hunk 2 above)
Comment (akankshamahajan15) [reply tools/db_crashtest.py:238]: Yes.

File: db_stress_tool/db_stress_gflags.cc (hunk 1 above)
Comment (akankshamahajan15) [reply db_stress_tool/db_stress_gflags.cc:601]: Yes. Let me update the description

**Review by anand1976 — APPROVED** [review]

**Pull request closed without merging**